File: src/345. Reverse Vowels of a String.js
/**
 * @param {string} s
 * @return {string}
 */

export function reverseVowels(s) {
  const arr = s.split(''),
    vowels = 'aeiouAEIOU'
  let left = 0,
    right = arr.length - 1
  while (left < right) {
    while (!vowels.includes(arr[left]) && left < right) left++
    while (!vowels.includes(arr[right]) && left < right) right--
    ;[arr[left], arr[right]] = [arr[right], arr[left]]
    left++
    right--
  }
  return arr.join('')
}

File: src/345. Reverse Vowels of a String.cpp
class Solution {
private:
    bool isVowel(char c) {
        return c == 'a' ||
            c == 'e' ||
            c == 'i' ||
            c == 'o' ||
            c == 'u' ||
            c == 'A' ||
            c == 'E' ||
            c == 'I' ||
            c == 'O' ||
            c == 'U';
    }
public:
    string reverseVowels(string s) {
        int left = 0,
            right = s.size() - 1;
        while (left < right) {
            while (!isVowel(s[left]) && left < right) left++;
            while (!isVowel(s[right]) && left < right) right--;
            swap(s[left], s[right]);
            left++;
            right--;
        }
        return s;
    }
};